uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -192,6 +193,8 @@ define([
         //check if user has changed any attribute
         if (this.isMode('edit')) {
           canSubmit = canSubmit && !_.isEqual(initialLeaveRequestAttributes, this.request.attributes());
+          //user has added a comment
+          canSubmit = canSubmit || this.request.comments.length != originalCommentsLength;
```

Comment:
You are checking the comments length only in edit mode. But comments can be added in view mode also. That is not handled.